src/Microsoft.PowerShell.ConsoleHost/host/msh/ManagedEntrance.cs

```diff
@@ -30,19 +32,39 @@ public sealed class UnmanagedPSEntry
         /// <param name="argc">
         /// Length of the passed in argument array.
         /// </param>
+        [Obsolete("Callers should now use UnmanagedPSEntry.Start(string[])", error: true)]
         public static int Start(string consoleFilePath, [MarshalAs(UnmanagedType.LPArray, ArraySubType = UnmanagedType.LPWStr, SizeParamIndex = 2)]string[] args, int argc)
         {
+            return Start(args);
+        }
+
+        /// <summary>
+        /// Starts managed MSH.
+        /// </summary>
+        /// <param name="args">
+        /// Command line arguments to the managed MSH
+        /// </param>
+        public static int Start(string[] args)
+        {
+            if (args == null)
+            {
+                throw new ArgumentNullException(nameof(args));
+            }
+
             // Warm up some components concurrently on background threads.
             EarlyStartup.Init();
 
-            // We need to read the settings file before we create the console host
-            CommandLineParameterParser.EarlyParse(args);
+            var banner = string.Format(
+                CultureInfo.InvariantCulture,
+                ManagedEntranceStrings.ShellBannerNonWindowsPowerShell,
+                PSVersionInfo.GitCommitId);
```

Comment:
The handling of `-isswait` should be moved before `ConsoleHost.ParseCommandLine(args)`, so that I can debug the arg parsing code.

The `Thread.CurrentThread.CurrentUICulture = NativeCultureResolver.UICulture;` code should also be moved before arg parsing, so the change of `CultureInfo` is respected when generating error message. I think `NativeCultureResolver` is still a stub today, but it's better to arrange the code in the way that it will work in future.

Reply:
Done.